src/utils/convert3Dto4D.m

```diff
@@ -0,0 +1,200 @@
+% (C) Copyright 2020 CPP BIDS SPM-pipeline developers
+
+function convert3Dto4D
+  %
+  % Short description of what the function does goes here.
+  %
+  % USAGE::
+  %
+  %   [argout1, argout2] = templateFunction(argin1, [argin2 == default,] [argin3])
+  %
+  % :param argin1: (dimension) obligatory argument. Lorem ipsum dolor sit amet,
+  %                consectetur adipiscing elit. Ut congue nec est ac lacinia.
+  % :type argin1: type
+  % :param argin2: optional argument and its default value. And some of the
+  %               options can be shown in litteral like ``this`` or ``that``.
+  % :type argin2: string
+  % :param argin3: (dimension) optional argument
+  %
+  % :returns: - :argout1: (type) (dimension)
+  %           - :argout2: (type) (dimension)
+  %
+  % .. todo:
+  %
+  %    - expand to run through multiple subjs ans groups
+  %      (https://stackoverflow.com/questions/8748976/list-the-subfolders-in-a-folder-matlab-only-subfolders-not-files)
+  %    - generalize how to retrieve RT from sidecar json file
+  %    - saveMatlabBatch(matlabbatch, ['3Dto4D_dataType-' num2str(dataType) '_RT-' num2str(RT)], opt, subID);
+  
+  % Add folder where some functions live
+  initEnv()
+
+  % Set the folder where sequences folders exist
+  optSource.sourceDir = '/Users/barilari/Desktop/DICOM_UCL_leuven/renamed/sub-pilot001/ses-002/MRI';
+
+  % List of the sequences that you want to skip (folder name pattern)
+  optSource.sequenceToIgnore = {'AAHead_Scout', ...
+                                'b1map', ...
+                                't1', ...
+                                'gre_field'};
+
+  % Number of volumes to discard ad dummies, (0 is default)
+  optSource.nbDummies = 5;
+
+  % List of the sequences where you want to remove dummies (folder name pattern)
+  optSource.sequenceRmDummies = {'cmrr_mbep2d_p3_mb2_1.6iso_AABrain', ...
+                                 'cmrr_mbep2d_p4_mb2_750um_AAbrain'};
+
+  % Set data format conversion (0 is default)
+
+  % 0:  SAME
+  % 2:  UINT8   - unsigned char
+  % 4:  INT16   - signed short
+  % 8:  INT32   - signed int
+  % 16: FLOAT32 - single prec. float
+  % 64: FLOAT64 - double prec. float
+
+  optSource.dataType = 0;
+
+  % Boolean to enable gzip of the new 4D file (0 is default)
+  optSource.zip = 0;
```

Comment:
Doesn't it make more sense to have all of those declared in a script and then passed to `convert3Dto4D` as an argument?

I think we should minimize how much users have to go and open functions and modify things in the source folder.

Reply:
Now that there are so many inputs to tweak yes, definetly